File: CSharp/ByTier/A/2139_A_Maple_and_Multiplication.cs
// Problem: CF 2139 A - Maple and Multiplication
// https://codeforces.com/contest/2139/problem/A

/*
 * Problem: A. Maple and Multiplication
 * Purpose: Determine the minimum number of operations to make two positive integers a and b equal,
 *          where each operation allows multiplying either a or b by any positive integer.
 * 
 * Algorithms/Techniques:
 *   - Mathematical analysis based on divisibility and equality conditions.
 *   - Greedy approach: If a equals b, 0 operations. If one divides the other, 1 operation.
 *     Otherwise, 2 operations are needed.
 * 
 * Time Complexity: O(1) per test case — constant time operations (no loops or recursion).
 * Space Complexity: O(1) — uses only a fixed amount of memory regardless of input size.
 */

using System;
class Program {
  static void Main() {
    int t = int.Parse(Console.ReadLine()); // Read number of test cases
    while (t-- > 0) { // Process each test case
      var input = Console.ReadLine().Split(); // Read and split the input line
      long a = long.Parse(input[0]); // Parse first number
      long b = long.Parse(input[1]); // Parse second number
      int result;
      if (a == b)
        result = 0; // No operations needed if a equals b
      else if (a % b == 0 || b % a == 0)
        result = 1; // One operation if one divides the other
      else
        result = 2; // Two operations otherwise
      Console.Write(result); // Output the result
      if (t > 0)
        Console.WriteLine(); // Print newline between test cases (except after last)
    }
  }
}


// https://github.com/VaHiX/codeForces/

File: Kotlin/ByTier/A/2139_A_Maple_and_Multiplication.kt
// Problem: CF 2139 A - Maple and Multiplication
// https://codeforces.com/contest/2139/problem/A

/*
 * Problem: A. Maple and Multiplication
 * Purpose: Determine the minimum number of operations to make two positive integers a and b equal.
 *          Each operation allows multiplying either a or b by any positive integer.
 *
 * Algorithms/Techniques:
 * - Mathematical analysis of divisibility and equality conditions
 * - Greedy approach based on modular arithmetic checks
 *
 * Time Complexity: O(1) per test case (constant time operations)
 * Space Complexity: O(1) (only using a fixed amount of extra space)
 */

fun main() {
    var v = readLine()!!.toInt() // Read number of test cases
    while(v > 0) {
        val input = readLine()!! // Read the two integers a and b
        val parts = input.split(" ") // Split input string into parts
        val a = parts[0].toInt() // First integer
        val b = parts[1].toInt() // Second integer
        if(a == b) {
            println(0) // Already equal, no operations needed
        } else if(a % b == 0 || b % a == 0) {
            println(1); // One is divisible by the other, one operation suffices
        } else {
            println(2); // Neither divides the other, two operations required
        }
        v--; // Decrement counter for next test case
    }
}


// https://github.com/VaHiX/codeForces/